src/Config/ObjectModel/RuntimeConfig.cs

```diff
@@ -44,30 +47,96 @@ public IEnumerable<DataSource> ListAllDataSources()
-    /// <param name="Schema">schema.</param>
+    /// <param name="Schema">schema for config.</param>
     /// <param name="DataSource">Default datasource.</param>
     /// <param name="Runtime">Runtime settings.</param>
     /// <param name="Entities">Entities</param>
+    /// <param name="DataSourceFiles">List of datasource files for multiple db scenario. Null for single db scenario.</param>
     [JsonConstructor]
-    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities)
+    public RuntimeConfig(string Schema, DataSource DataSource, RuntimeOptions Runtime, RuntimeEntities Entities, DataSourceFiles? DataSourceFiles = null)
     {
         this.Schema = Schema;
         this.DataSource = DataSource;
         this.Runtime = Runtime;
         this.Entities = Entities;
-        this._dataSourceNameToDataSource = new Dictionary<string, DataSource>();
-        this._defaultDataSourceName = Guid.NewGuid().ToString();
-        this._dataSourceNameToDataSource.Add(this._defaultDataSourceName, this.DataSource);
+        _defaultDataSourceName = Guid.NewGuid().ToString();
+
+        // we will set them up with default values
+        _dataSourceNameToDataSource = new Dictionary<string, DataSource>
+        {
+            { _defaultDataSourceName, this.DataSource }
+        };
 
-        this._entityNameToDataSourceName = new Dictionary<string, string>();
+        _entityNameToDataSourceName = new Dictionary<string, string>();
         foreach (KeyValuePair<string, Entity> entity in Entities)
         {
-            _entityNameToDataSourceName.TryAdd(entity.Key, this._defaultDataSourceName);
+            _entityNameToDataSourceName.TryAdd(entity.Key, _defaultDataSourceName);
+        }
+
+        // Process data source and entities information for each database in multiple database scenario.
+        this.DataSourceFiles = DataSourceFiles;
+
+        if (DataSourceFiles is not null && DataSourceFiles.SourceFiles is not null)
+        {
+            IEnumerable<KeyValuePair<string, Entity>> allEntities = Entities.AsEnumerable();
+            // Iterate through all the datasource files and load the config.
+            IFileSystem fileSystem = new FileSystem();
+            FileSystemRuntimeConfigLoader loader = new(fileSystem);
+
+            foreach (string dataSourceFile in DataSourceFiles.SourceFiles)
+            {
+                if (loader.TryLoadConfig(dataSourceFile, out RuntimeConfig? config))
+                {
+                    try
+                    {
+                        _dataSourceNameToDataSource = _dataSourceNameToDataSource.Concat(config._dataSourceNameToDataSource).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        _entityNameToDataSourceName = _entityNameToDataSourceName.Concat(config._entityNameToDataSourceName).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                        allEntities = allEntities.Concat(config.Entities.ToList());
```

Comment:
to be consistent with line 83, why not us `config.Entities.AsEnumerable()`?

Reply:
fixed.